### PR TITLE
Refactor layer-wise tests with torch

### PR DESCRIPTION
**Pull request by hengtaoguo**
# Description

Reverts the standalone `cpu_torch_reference_tests` workflow and integrates `gemma4_layers_test.py` into the standard `cpu-unit` tests. The test is configured to run only on scheduled CI runs.
- Removed `maxtext_cpu_torch_reference_tests` workflow.
- Enabled `gemma4_layers_test.py` in `pytest.ini`, and marked it as `cpu_only` and `scheduled_only`.
- Configured torch installation for `cpu-unit` only during scheduled runs.

**TODO**: After this atomic PR, follow up with another PR refactoring `import torch` for other layer-wise tests, and add them to scheduled runs cleanly.

# Tests

```
JAX_PLATFORMS=cpu python -m pytest tests/unit/gemma4_layers_test.py
```

**Note**: GitHub Actions Scan is failing due to a very recent Google-wise security rollout on [07-27-2026](https://screenshot.googleplex.com/5jRDKaaYxvQo7ts).

Manual work-dispatch run (simulating scheduled runs) with torch installation: [successful](https://screenshot-v2.corp.google.com/3q7Trj3zWz5cC9f)
# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: .github/workflows/run_tests_coordinator.yml [hunk 1]
```diff
@@ -231,4 +224,4 @@ jobs:
       total_workers: ${{ fromJSON(needs.setup-parameters.outputs.total_workers) }}
       maxtext_sha: ${{ inputs.maxtext_sha }}
       is_update_hlo: ${{ inputs.is_update_hlo }}
-      install_torch_cpu: ${{ inputs.flavor == 'cpu-torch-reference' }}
+      install_torch_cpu: ${{ inputs.flavor == 'cpu-unit' && inputs.is_scheduled_run }}
```
Comment (SurbhiJainUSC) [inline .github/workflows/run_tests_coordinator.yml:227]: Won't this install `torch` for all CPU + scheduled tests, which are not even torch ref tests?

Comment (hengtaoguo) [reply .github/workflows/run_tests_coordinator.yml:227]: Yes it is, since we are merging the layer-wise tests into the typical `cpu-unit` test category. 

The only way to separate this was to follow the old format using a separate list called `cpu-torch-reference-test`, but it may introduce more complexity: now we only have one source-of-truth in the [ignore list](https://github.com/AI-Hypercomputer/maxtext/blob/f54d888feb611cbf15b8924d8e9f9b5e6a0ca6cf/pytest.ini#L4).

The torch installation process takes extra ~5s, which is marginal to me especially it is now schedule-only.

Comment (SurbhiJainUSC) [reply .github/workflows/run_tests_coordinator.yml:227]: I see `torch` is installed without deps: https://github.com/AI-Hypercomputer/maxtext/blob/0834d373fb895c966c9c5b018c97c2f000365b6b/.github/workflows/run_tests_against_package.yml#L132. So, that should not be the problem. I was just wondering if installing `torch` would change our deps set, but `--no-deps` would ensure that it is not changed. So, we are good.

**Review by SurbhiJainUSC — APPROVED** [review]

**Review by YixuanWang-99 — APPROVED** [review]

**Comment by hengtaoguo** [comment]
New scheduled tests today successfully picked up this test, and all passed:

https://screenshot-v2.corp.google.com/4bezb9desgu5Ldj
